### PR TITLE
Feature/fetch acl sample

### PR DESCRIPTION
**Pull request by chasinandrew**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [X] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [X] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by chasinandrew** [comment]
Looks like the project connected to the unit test has universal access, so no ACL will be fetched. Can we change the project that runs in the CI build to one that has ACL?

**Comment by chasinandrew** [comment]
@averikitsch @telpirion @holtskinner would any of you be able to help review this? Also is it possible to change the project ID in the CI pipeline? The one currently there does not have any ACL so the test is failing.

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:95]: Please add a new line.

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:34]:
```suggestion
  public static void main(String[] args) throws IOException {
```
The format is to use a main method and then call the snippet.

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:39]: Can you kindly provide more details about these variables and possibly a link to docs?

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:43]: Please add a method comment in line 42, explaining the goal of this method.

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:51]: This comment should be added before the client initialization.
```suggestion
    // Initialize client that will be used to send requests. This client only needs to be
    // created once, and can be reused for multiple requests.
    try (DocumentServiceClient documentServiceClient = 
```

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:87]:
```suggestion
    // Initialize client that will be used to send requests. This client only needs to be
    // created once, and can be reused for multiple requests.
    try (ProjectsClient projectsClient = ProjectsClient.create()) { 
```

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:81]: Instead of sending FetchAclRequest in the if else condition, wdyt about using the condition to create a new resource string and then outside the block, create the request?

Something like, 

```
String resource = "";

if (condition) {
 resource = documentName.toString();
} else {
 resource = projectNumber
}

.setResource(resource);
```

File: content-warehouse/src/test/java/contentwarehouse/v1/FetchAclTest.java
Comment (Sita04) [inline content-warehouse/src/test/java/contentwarehouse/v1/FetchAclTest.java:33]:
```suggestion
@RunWith(JUnit4.class)
public class FetchAclTest {
```

**Review by Sita04 — CHANGES_REQUESTED** [review]
Thanks for the sample!
PTAL at the comments and also the test failures.

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (chasinandrew) [reply content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:85]: I'm not sure that this would work due to .setProjectOwner(true), it is only applicable when the ACL request is for a project. It would throw an error if it's set on the document scope.

File: content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java
Comment (chasinandrew) [reply content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java:47]: Done, please let me know if my description is sufficient

**Comment by chasinandrew** [comment]
Pull request below will unblock this PR once merged. 
https://github.com/GoogleCloudPlatform/java-docs-samples/pull/8461

**Comment by chasinandrew** [comment]
Closing this PR as this sample is deprioritized

**Pull request closed without merging**
